lib/src/arithmetic/divider.dart

```diff
@@ -0,0 +1,362 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// divider.dart
+// Implementation of Integer Divider Module.
+//
+// 2024 August
+// Author: Josh Kimmel <joshua1.kimmel@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// State object for Divider processing.
+class _MultiCycleDividerState {
+  /// Width of state in bits.
+  static int width = 3;
+
+  /// Ready for a new division.
+  static Const ready = Const(0, width: width);
+
+  /// Processing a current step in the algorithm.
+  static Const process = Const(1, width: width);
+
+  /// Accumulating the result of a current step in the algorithm.
+  static Const accumulate = Const(2, width: width);
+
+  /// Converting the final result of the algorithm.
+  static Const convert = Const(3, width: width);
+
+  /// Division complete.
+  static Const done = Const(4, width: width);
+}
+
+/// Internal interface to the Divider.
+class MultiCycleDividerInterface extends PairInterface {
+  /// Clock for sequential logic.
+  Logic get clk => port('clk');
+
+  /// Reset for sequential logic (active high).
+  Logic get reset => port('reset');
+
+  /// Dividend (numerator) for the division operation.
+  Logic get dividend => port('dividend');
+
+  /// Divisor (denominator) for the division operation.
+  Logic get divisor => port('divisor');
+
+  /// The integrating environment is ready to accept the output of the module.
+  Logic get readyOut => port('readyOut');
+
+  /// Request for a new division operation to be performed.
+  Logic get validIn => port('validIn');
+
+  /// Quotient (result) for the division operation.
+  Logic get quotient => port('quotient');
+
+  /// A Division by zero occurred.
+  Logic get divZero => port('divZero');
+
+  /// The result of the current division operation is ready.
+  Logic get validOut => port('validOut');
+
+  /// The module is ready to accept new inputs.
+  Logic get readyIn => port('readyIn');
+
+  /// The width of the data operands and result.
+  final int dataWidth;
+
+  /// A constructor for the divider interface.
+  MultiCycleDividerInterface({this.dataWidth = 32})
+      : super(portsFromProvider: [
+          Port('clk'),
+          Port('reset'),
+          Port('dividend', dataWidth),
+          Port('divisor', dataWidth),
+          Port('validIn'),
+          Port('readyOut'),
+        ], portsFromConsumer: [
+          Port('quotient', dataWidth),
+          Port('divZero'),
+          Port('validOut'),
+          Port('readyIn'),
+        ]);
+
+  /// A match constructor for the divider interface.
+  MultiCycleDividerInterface.match(MultiCycleDividerInterface other)
+      : this(dataWidth: other.dataWidth);
+}
+
+/// The Divider module definition
+class MultiCycleDivider extends Module {
+  /// The Divider's interface declaration.
+  late final MultiCycleDividerInterface intf;
+
+  /// Get interface's validOut signal value.
+  Logic get validOut => intf.validOut;
```

Comment:
In concept I understand what you mean, but I'm not sure what this looks like mechanically in ROHD.

Reply:
Keeping a copy of the external interface seems annoying, alternatively you could just call `output('nameOfPort')` knowing that the port punched on the module would match the interface name

Reply:
I just realized making the interface protected breaks a lot of stuff in the test where the interface is accessed directly. How should I deal with this? Create getters for the inputs as well and use those?

Reply:
A couple options:
- Pass the original (external) interface to your testbench instead of the DUT itself.  This mimics how a "user" of the Divider would interact with it. [recommended]
- Just ignore the lints as you're testing internal behavior of the divider.

The inputs to a module should also be protected (not used outside the module).